pennylane/grouping/group_observables.py

```diff
@@ -224,7 +225,7 @@ def group_observables(observables, coefficients=None, grouping_type="qwc", metho
     """
 
     if coefficients is not None:
-        if len(coefficients) != len(observables):
+        if qml.math.shape(coefficients)[0] != len(observables):
```

Comment:
I haven't tested yet whether this rewrite in interface-agnostic manner is differentiable...

Reply:
It is not, but I will fix it when implementing things properly - should not be a blocker.